webdriver/tests/element_click/link.py

```diff
@@ -0,0 +1,85 @@
+import pytest
+
+from tests.support.asserts import assert_error, assert_success
+from tests.support.inline import inline
+
+def click(session, element):
+    return session.transport.send("POST", "session/{session_id}/element/{element_id}/click"
+                                  .format(session_id=session.session_id,
+                                          element_id=element.id))
+
+
+# 14.1 Element Click Link Element Tests Step 7
+
+def test_click_numbers_link(session):
+    # 14.1 Step 7
+    url = "/webdriver/tests/element_click/support/input.html"
+    session.url = inline("<a href=%s>123456</a>" % url)
+    element = session.find.css("a", all=False)
+    response = click(session, element)
+    assert_success(response)
+
+    assert session.url == "http://web-platform.test:8000%s" % url
+
+
+def test_click_multi_line_link(session):
+    # 14.1 Step 7
+    url = "/webdriver/tests/element_click/support/input.html"
+    session.url = inline("""
+        <p style="background-color: rgb(255, 255, 0); width: 5em;">
+            <a href=%s>Helloooooooooooooooooooo Worlddddddddddddddd</a>
+        </p>""" % url)
+    element = session.find.css("a", all=False)
+    response = click(session, element)
+    assert_success(response)
+
+    assert session.url == "http://web-platform.test:8000%s" % url
+
+
+def test_click_link_unload_event(session):
+    # 14.1 Step 7
+    url = "/webdriver/tests/element_click/support/input.html"
+    session.url = inline("""<body onunload="checkUnload()">
+                                <a href=%s>click here</a>
+                                <input type=checkbox>
+                                <script>
+                                function checkUnload() {
+                                    document.getElementsByTagName("input")[0].checked = true;
+                                }
+                                </script>
+                            </body>""" % url)
+    element = session.find.css("a", all=False)
+    response = click(session, element)
+    assert_success(response)
+
+    assert session.url == "http://web-platform.test:8000%s" % url
+
+    session.back()
+
+    element = session.find.css("input", all=False)
+    response = session.execute_script("return document.getElementsByTagName(\"input\")[0].checked;")
+    assert response is True
+
+
+def test_click_link_hash(session):
+    # 14.1 Step 7
+    session.url = inline("""<a href="#">aaaa</a>""")
```

Comment:
Can you clarify this? I am not 100% sure I understand your question.

Reply:
Since this anchor is empty, clicking it won’t take you anywhere. It will neither navigate to another document nor scroll to a different position in the document.

Can we add another test that _does_ scroll? Preferably I think the anchor should be located _way outside_ the viewport so the scrolling takes a while, e.g. `<p id=anchor style="margin-top: 5000vh">scroll here`.

Reply:
There is a scroll test in element_click.py. Do you want me to add another scroll test where clicking a link will make it scroll?

Reply:
This would explicitly be testing scrolling behaviour, but that the driver doesn’t navigate on clicking an anchor.

Reply:
added